### PR TITLE
Added SAMOVE command

### PR DESCRIPTION
**Pull request by hoerup**

### CODE REVIEW

File: 3.0/m_samove.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	{
+		if (parameters.size() != 3) 
+		{
+			return CMD_FAILURE;			
+		}
+
+		const std::string& nickname = parameters[0];
+		const std::string& from_channel = parameters[1];
+		const std::string& to_channel = parameters[2];
+		std::string msg = std::string();
```
Comment (SadieCat) [inline 3.0/m_samove.cpp:59]: Declaring msg here is unnecessary as you can use an empty string literal when calling PartUser.

File: 3.0/m_samove.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+			Channel* from_chan = ServerInstance->FindChan(from_channel);
+			if (!(from_chan))
+			{
+				user->WriteRemoteNotice("*** invalid 'from channel '" + from_channel);
+				return CMD_FAILURE;
+			}
+			Channel* to_chan = ServerInstance->FindChan(to_channel);
+			if (!(to_chan))
+			{
+				user->WriteRemoteNotice("*** invalid 'to channel '" + to_channel);
```
Comment (SadieCat) [inline 3.0/m_samove.cpp:91]:
```suggestion
				user->WriteRemoteNotice("*** invalid 'to channel' " + to_channel);
```

File: 3.0/m_samove.cpp [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	CommandSamove cmd;
+ public:
+	ModuleSamove()
+		: cmd(this)
+	{
+	}
+
+	Version GetVersion() CXX11_OVERRIDE
+	{
+		return Version("Adds the /SAMOVE command which allows server operators to force move users from one channel to another.");
```
Comment (SadieCat) [inline 3.0/m_samove.cpp:160]:
```suggestion
		return Version("Adds the /SAMOVE command which allows server operators to force move users from one channel to another.", VF_OPTCOMMON);
```

File: 3.0/m_samove.cpp (hunk 1 above)
Comment (hoerup) [reply 3.0/m_samove.cpp:59]: unfortunately Channel::PartUser() doesn't accept a const string reference ( i think thats a bug?)

File: 3.0/m_samove.cpp (hunk 2 above)
Comment (hoerup) [reply 3.0/m_samove.cpp:91]: done

File: 3.0/m_samove.cpp (hunk 3 above)
Comment (hoerup) [reply 3.0/m_samove.cpp:160]: done

File: 3.0/m_samove.cpp (hunk 1 above)
Comment (SadieCat) [reply 3.0/m_samove.cpp]: Ah, that is a bug yeah.

For now just move the declaration next to the function cal and remove the unnecessary value (i.e. `std::string msg;`).

Comment (hoerup) [reply 3.0/m_samove.cpp]: Thats also what i dit in my latest commit

**Pull request merged**